hooks/useTranslations.tsx

```diff
@@ -0,0 +1,7 @@
+import { t } from 'lib/translate'
+
+export const useTranslations = (lang: string) => {
+  return function (key: string) {
+    return t(key, lang)
+  }
+}
```

Comment:
Unfortunately that won't work, my aim was also to make `useTranslations` work on both client- and server-side but it would seem Next 13 w/ the app dir approach doesn't have a way to get the locale other than prop drilling it to where you need it for Server components.

This is why I split up my original logic into `useTranslations` and `useLang`, where `useLang` could be used for Client components (reading locale from url) and Server components would get it from the route params passed by Next.

I'll admit its not ideal, but I think we should leave this for now. Though it's a bit inconvenient, it does work. We can (and should) refactor when Vercel comes up with an official i18n solution for the app dir approach.